agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/ResolveMarchOrderGameState.ts

```diff
@@ -18,8 +18,13 @@ import TakeControlOfEnemyPortGameState, { SerializedTakeControlOfEnemyPortGameSt
 import { findOrphanedShipsAndDestroyThem } from "../../port-helper/PortHelper";
 
 export default class ResolveMarchOrderGameState extends GameState<ActionGameState, ResolveSingleMarchOrderGameState | CombatGameState | TakeControlOfEnemyPortGameState> {
+    public currentTurnOrderIndex: number;
+
     constructor(actionGameState: ActionGameState) {
         super(actionGameState);
+
+        // Start at -1 to indicate the GameState didn't start iterating yet.
+        this.currentTurnOrderIndex = -1;
```

Comment:
Can you put this in `firstStart`? You can even remove this constructor as the parent one is exactly the same.

Reply:
I don't mind too much, but wouldn't this cause issues if we want to call `getNextHouseToResolveMarchOrder` without calling `firstStart` before hand? I'd believe such a situation could arise in a testing environment

Reply:
`firstStart` should always be called after instantiating a GameState and the references have been properly set up (`parentGameState`, ...). It's kind of a convention in the project to have these kind of methods on GameStates.